### PR TITLE
Make this work on MacOS

### PR DESCRIPTION
**Pull request by lodenrogue**
I love this little tool. I think its very useful. When I went to run the project on MacOS I had to make a few changes to get it to work.

1. Mac has curl installed by default not wget
So I added an if condition to use curl if wget is not found

2. The go version needed to be updated

3. The extra comma at the end of the json was throwing an exception so I removed it

Please let me know if you would like any changes or have any questions.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sgrep
 
-go 1.22.5
+go 1.23
```
Comment (arunsupe) [inline go.mod]: Not sure about this change. go's current stable version is 1.22.5. You must have the unstable bleeding edge dev version?

**Review by arunsupe — CHANGES_REQUESTED** [review]
Not sure about changing to go version 1.23. Latest stable go version is 1.22.5. Please change back. 

Everything else looks good. Once version is changed back, will approve.

And thanks for making it work on a mac. I only have a PC. Grateful for your help.

**Comment by lodenrogue** [comment]
Reverted go version.

My pleasure. If you need any other help let me know!

**Comment by lodenrogue** [comment]
If you're on PC I would test the `2 > /dev/null` to make sure that works. Its a unix thing so not sure how windows would handle that.

**Review by arunsupe — APPROVED** [review]

**Pull request merged**